What's New for GRC 2.14 (#7878)
open-cluster-management/rhacm-docs: release_notes/acm_whats_new.adoc

[#governance-whats-new]
== Governance

> * To convert YAML strings, implement the `fromYaml` and `toYaml` Go template functions in the `ConfigurationPolicy` resource. You also have more Sprig functions in the list of available functions.

Reviewer: @JustinKuli thanks! I wanted to diversify the sentence structure so it's not always "You can now.." let me know what you think. 

Also, when I say links to Docs, I mean the doc that has been updated due to these recent changes. For example, see the Governance What's New from 2.13: https://github.com/stolostron/rhacm-docs/blob/2.13_stage/release_notes/acm_whats_new.adoc 

If there aren't any docs in 2.14 that have been modified or changed, that's fine, just let me know :)

Author: Here are the doc sections:
- fromYaml / toYaml : https://github.com/stolostron/rhacm-docs/blob/2.14_stage/governance/template_functions.adoc#fromYaml-function
- dryrun / policytools CLI : will be documented at https://github.com/stolostron/rhacm-docs/blob/2.14_stage/governance/manage_policies.adoc#policy-command-line-interface via https://issues.redhat.com/browse/ACM-21150
- the related resources UI doesn't have any relevant documentation
- versions list in OperatorPolicy: https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.13/html-single/governance/index#operator-policy-yaml-table via https://issues.redhat.com/browse/ACM-21152

== Governance

- You can now use the `fromYaml` and `toYaml` Go template functions in the `ConfigurationPolicy` resource to convert YAML strings. You also have more Sprig functions in the list of available functions.

> - You can now use the `dryrun` subcommand in the `policytools` CLI to evaluate a `ConfigurationPolicy` locally using static resource files. This subcommand is now Generally Available.

Reviewer: @JustinKuli thanks! This reads really well, but I think I can make a couple of small tweaks. I can make those in a commit. 

In a recent PR I had for the "what's new" I was asked to include links to the files that entailed these changes and updates. Do you mind sending me those files, and I can link to them from these entries?

Author: Do you mean the dev stories? Or are you looking for other information?
- For the versions list: https://issues.redhat.com/browse/ACM-20804
- For the UI change: https://issues.redhat.com/browse/ACM-20098
- For the dryrun CLI, I'm not really sure what story is best to link for information... The original story was https://issues.redhat.com/browse/ACM-14161, back when it was going to be Dev Preview for 2.12. I don't know if that's particularly useful.

Of course, I'm fine with reviewing any improvements you see

[#governance-whats-new]
== Governance

> - You can now use the `.Object` Go template context variable in the `ConfigurationPolicy` resource to reference values from the target object on the cluster, alongside the `.ObjectName` and `.ObjectNamespace` context variables.

Author: This section about the .Object variable must have gotten lost during a merge, I think we'll need to put it back

> * To convert YAML strings, implement the `fromYaml` and `toYaml` Go template functions in the `ConfigurationPolicy` resource. You also have more Sprig functions in the list of available functions. For more information, see link:../governance/template_functions.adoc#fromYaml-function[_fromYaml_].

Reviewer: ```suggestion
* You can now use the `.Object` Go template context variable in the `ConfigurationPolicy` resource to reference values from the target object on the cluster, alongside the `.ObjectName` and `.ObjectNamespace` context variables. For more information, see the link:../governance/template_support_intro.adoc#template-comparison-table[Comparison of hub cluster and managed cluster templates].
* To convert YAML strings, implement the `fromYaml` and `toYaml` Go template functions in the `ConfigurationPolicy` resource. You also have more Sprig functions in the list of available functions. For more information, see link:../governance/template_functions.adoc#fromYaml-function[_fromYaml_].
```

* With the `dryrun` subcommand being available, you can run it in the `policytools` CLI to evaluate a `ConfigurationPolicy` locally by accessing the static resource files. For more information, see link:../governance/manage_policies.adoc#policy-command-line-interface[Policy command line interface]. 

> * You can now view related resources for your policies across different clusters in the `Discovered policies` section in the console from one table, instead of needing to go to the policy template for each cluster. This table works for all the policy types except for the {gate} resources.

Author: Is `{gate}` correct? Does that expand to Gatekeeper?

Reviewer: Renders to `Gatekeeper operator` let me know if you're concerned with that.

[#governance-whats-new]
== Governance

* You can now use the `.Object` Go template context variable in the `ConfigurationPolicy` resource to reference values from the target object on the cluster, alongside the `.ObjectName` and `.ObjectNamespace` context variables. For more information, see the link:../governance/template_support_intro.adoc#template-comparison-table[Comparison of hub cluster and managed cluster templates].
* To convert YAML strings, implement the `fromYaml` and `toYaml` Go template functions in the `ConfigurationPolicy` resource. You also have more Sprig functions in the list of available functions. For more information, see link:../governance/template_functions.adoc#fromYaml-function[_fromYaml_].
* With the `dryrun` subcommand being available, you can run it in the `policytools` CLI to evaluate a `ConfigurationPolicy` locally by accessing the static resource files. For more information, see link:../governance/manage_policies.adoc#policy-command-line-interface[Policy command line interface]. 

> * You can now view related resources for your policies across different clusters in the `Discovered policies` section in the console from one table, instead of needing to go to the policy template for each cluster. This table works for all the policy types except for the {gate} resources.

Reviewer 2: Based on the IBM Style Guide, this needs to be more concise; describing the previous user step is not required. And "Discover policies" should be Italicized instead of monospaced. Discovered policies are mentioned in the following docs: **Policy deployment with external tools and Policy deployment**

Seems like we should point to one of those. Maybe Policy deployment. Please let me know your thoughts about this rewrite before committing @JustinKuli @jc-berger 
```suggestion
* You can now view related resources for your policies across different clusters within a single table in the _Discovered policies_ section of the console. This table works for all the policy types except for the {gate} resources. See link:../governance/policy_deployment.adoc#policy-deployment[Policy deployment] for more details.
```

Author: I guess I don't know what the purpose of a What's New section is, if you can't contrast the new feature with what was previously available.

Reviewer 2: If users want to know what was released in the previous version, it is assumed that the user will navigate to the What's new of the previous version (or whatever version they are on). The focus is the new behavior, plus this helps with keeping content concise.

Author: I'm fine with it either way

> * You can now view related resources for your policies across different clusters within a single table in the _Discovered policies_ section of the console. This table works for all the policy types except for the {gate} resources. See link:../governance/policy_deployment.adoc#policy-deployment[Policy deployment] for more details.

Reviewer 2: @jc-berger do you agree with my change here? Apologies. I made the commit before you responded

> * To give different operator versions in one entry of the `versions` list inside an `OperatorPolicy`, separate these versions with a comma. For more information, see link:../governance/operator_pol_ctrl.adoc#operator-policy-yaml-table[Operator policy YAML table]. 

Reviewer 2: Sorry i missed this update. 
```suggestion
* To give different operator versions in one entry of the `versions` list within an `OperatorPolicy` resource, separate the versions with a comma. For more information, see link:../governance/operator_pol_ctrl.adoc#operator-policy-yaml-table[Operator policy YAML table]. 
```
